source/_addonStore/dataManager.py

```diff
@@ -157,24 +181,24 @@ def getLatestCompatibleAddons(
 			self,
 			onDisplayableError: Optional[DisplayableError.OnDisplayableErrorT] = None,
 	) -> "AddonGUICollectionT":
+		cacheHash = self._getCacheHash()
 		shouldRefreshData = (
 			not self._compatibleAddonCache
 			or self._compatibleAddonCache.nvdaAPIVersion != addonAPIVersion.CURRENT
-			or _DataManager._cachePeriod < (datetime.now() - self._compatibleAddonCache.cachedAt)
+			or cacheHash and (self._compatibleAddonCache.cacheHash != cacheHash)

```

Comment:
I think we should always refresh the cache if fetching the cacheHash fails.
This assumes we can treat "None" as a valid cacheHash,

```suggestion
			or cacheHash is None
			or self._compatibleAddonCache.cacheHash != cacheHash
```